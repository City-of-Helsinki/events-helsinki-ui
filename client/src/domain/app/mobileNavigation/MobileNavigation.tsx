import React from "react";

import MobileHeader from "./mobileHeader/MobileHeader";
import styles from "./mobileNavigation.module.scss";
import MobileNavigationMenu from "./mobileNavigationMenu/MobileNavigationMenu";

const HeaderMobile: React.FC = () => {
  const [isMenuOpen, setIsmenuOpen] = React.useState(false);

  const handleCloseMenu = () => {
    setIsmenuOpen(false);
  };

  const handleOpenMenu = () => {
    setIsmenuOpen(true);
  };

  return (
    <div className={styles.mobileNavigation}>
      <MobileHeader
        isMenuOpen={isMenuOpen}
        onCloseMenu={handleCloseMenu}
        onOpenMenu={handleOpenMenu}
      />
      <MobileNavigationMenu
        isMenuOpen={isMenuOpen}
        onMenuClose={handleCloseMenu}
      />
    </div>
  );
};

export default HeaderMobile;
