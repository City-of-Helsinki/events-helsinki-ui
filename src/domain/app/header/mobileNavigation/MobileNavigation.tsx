import React from 'react';

import MobileMenu, { useMobileMenuContext } from '../mobileMenu/MobileMenu';
import MobileNavbar from '../mobileNavbar/MobileNavbar';
import styles from './mobileNavigation.module.scss';

const MobileNavigation: React.FC = () => {
  const { isMobileMenuOpen, toggleMenu } = useMobileMenuContext();

  const disabledBodyScrolling = () => {
    const body = document.body;

    body.classList.add('scrollDisabledOnMobile');
  };

  const enableBodyScrolling = () => {
    const body = document.body;

    body.classList.remove('scrollDisabledOnMobile');
  };

  React.useEffect(() => {
    if (isMobileMenuOpen) {
      disabledBodyScrolling();
    } else {
      enableBodyScrolling();
    }
  }, [isMobileMenuOpen]);

  return (
    <div className={styles.mobileNavigation}>
      <MobileNavbar isMenuOpen={isMobileMenuOpen} onToggleMenu={toggleMenu} />
      <MobileMenu isMenuOpen={isMobileMenuOpen} onClose={toggleMenu} />
    </div>
  );
};

export default MobileNavigation;
