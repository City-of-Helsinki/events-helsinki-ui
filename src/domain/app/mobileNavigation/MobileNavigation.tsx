import React from "react";

import MobileNavigation, {
  useMobileMenuContext
} from "../../../common/components/mobileMenu/MobileMenu";
import MobileHeader from "./mobileHeader/MobileHeader";
import styles from "./mobileNavigation.module.scss";

const HeaderMobile: React.FC = () => {
  const {
    closeMobileMenu,
    isMobileMenuOpen,
    openMobileMenu
  } = useMobileMenuContext();

  const disabledBodyScrolling = () => {
    const body = document.body;

    body.classList.add("scrollDisabledOnMobile");
  };

  const enableBodyScrolling = () => {
    const body = document.body;

    body.classList.remove("scrollDisabledOnMobile");
  };

  const handleCloseMenu = () => {
    closeMobileMenu();
    enableBodyScrolling();
  };

  const handleOpenMenu = () => {
    openMobileMenu();
    disabledBodyScrolling();
  };

  return (
    <div className={styles.mobileNavigation}>
      <MobileHeader
        isMenuOpen={isMobileMenuOpen}
        onCloseMenu={handleCloseMenu}
        onOpenMenu={handleOpenMenu}
      />
      <MobileNavigation
        isMenuOpen={isMobileMenuOpen}
        onClose={handleCloseMenu}
      />
    </div>
  );
};

export default HeaderMobile;
