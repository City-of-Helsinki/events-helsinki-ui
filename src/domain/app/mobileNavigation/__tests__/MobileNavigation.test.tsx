import { shallow } from 'enzyme';
import React from 'react';
import routeData, { MemoryRouter } from 'react-router';

import MobileNavigation from '../MobileNavigation';

const mockLocation = {
  hash: '',
  pathname: '/fi/home',
  search: '',
  state: '',
};
beforeEach(() => {
  jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
});

it('MobileNavigation matches snapshot', () => {
  const container = shallow(
    <MemoryRouter>
      <MobileNavigation />
    </MemoryRouter>
  );
  expect(container.html()).toMatchSnapshot();
});
