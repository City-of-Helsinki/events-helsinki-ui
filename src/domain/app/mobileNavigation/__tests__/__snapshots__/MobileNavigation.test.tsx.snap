// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavigation matches snapshot 1`] = `"<div class=\\"mobileNavigation\\"><header class=\\"mobileHeaderWrapper\\"><div class=\\"container\\"><div class=\\"mobileNavbar\\"><a aria-label=\\"Etusivu\\" class=\\"logoWrapper\\" href=\\"/\\"><div class=\\"logo\\"></div><div class=\\"appName\\">Tapahtumat</div></a><div class=\\"buttonWrapper\\"><button class=\\"menuButton\\" aria-label=\\"Avaa valikko\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><path d=\\"M1.01301075,4.01419355 L22.9875806,4.01419355 L22.9875806,6.61634409 L1.01301075,6.61634409 L1.01301075,4.01419355 Z M1.01301075,17.7482258 L22.9875806,17.7482258 L22.9875806,20.3503763 L1.01301075,20.3503763 L1.01301075,17.7482258 Z M1.01301075,10.8815054 L22.9875806,10.8815054 L22.9875806,13.4836559 L1.01301075,13.4836559 L1.01301075,10.8815054 Z\\"></path></svg></button></div></div></div></header><div aria-hidden=\\"true\\" class=\\"mobileMenu\\" style=\\"visibility:hidden\\"><div class=\\"linkWrapper\\"><ul><li class=\\"link\\"><a href=\\"/fi/events\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><path d=\\"M15.3582734,0.0535251799 C12.0567815,0.0517959979 9.04784082,1.94662012 7.62269829,4.92467659 C6.19755575,7.90273306 6.6096026,11.4345925 8.68201439,14.0046043 L0.00920863309,22.676259 L1.30820144,23.9758273 L9.98273381,15.3035971 C12.9168981,17.6690044 17.0515408,17.841428 20.1724602,15.7285309 C23.2933795,13.6156338 24.6689728,9.71271903 23.5625765,6.10989254 C22.4561801,2.50706606 19.1271543,0.0488985453 15.3582734,0.0517959979 L15.3582734,0.0535251799 Z M15.3582734,15.3657554 C11.637384,15.3657554 8.62100719,12.3493786 8.62100719,8.62848921 C8.62100719,4.90759984 11.637384,1.89122302 15.3582734,1.89122302 C19.0791628,1.89122302 22.0955396,4.90759984 22.0955396,8.62848921 C22.0917339,12.3479875 19.0777709,15.3624653 15.3582734,15.3669065 L15.3582734,15.3657554 Z\\"></path></svg>Etsi tekemistä</a></li><li class=\\"link\\"><a href=\\"/fi/collections\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g stroke-width=\\"1\\" fill-rule=\\"evenodd\\"><path d=\\"M11.9905843,1 L8.7985013,8.18485604 L1,9.03013322 L6.82418044,14.2980332 L5.21426319,22 L12.0080942,18.0692954 L18.8098542,21.9854148 L17.1857308,14.283448 L23,9.00593509 L15.2031506,8.18353012 L11.9905843,1 Z M8.67692331,13.692417 L4.93608746,10.3083249 L9.94721509,9.76535863 L11.9952096,5.15180263 L14.0603835,9.76668456 L19.062591,10.2950657 L15.3313361,13.6801522 L16.3746602,18.6294987 L12.0061119,16.1138875 L7.64218888,18.6381172 L8.67692331,13.692417 Z\\"></path></g></svg>Suosittelemme</a></li></ul></div><div class=\\"languageSelectWrapper\\"><ul><li class=\\"languageLink isSelected\\"><a lang=\\"fi\\" href=\\"/fi/home\\">Suomeksi</a></li><li class=\\"languageLink\\"><a lang=\\"sv\\" href=\\"/sv/home\\">På svenska</a></li><li class=\\"languageLink\\"><a lang=\\"en\\" href=\\"/en/home\\">In English</a></li></ul></div></div></div>"`;
